File: Blackjack21-JAVA/Blackjack21Library/src/Blackjack21/Game/Exceptions/InvalidHandSplitException.java
package Blackjack21.Game.Exceptions;

/**
 * Throws an exception if the hand cannot be split
 */
public class InvalidHandSplitException extends Exception
{

    public InvalidHandSplitException()
    {
        super("Invalid hand split, conditions to split the hand have not been met");
    }

    public InvalidHandSplitException(String message)
    {
        super(message);
    }

}

File: BlackJack-CSHARP/Blackjack21.Game/Game/Exceptions/InvalidHandSplitException.cs
using System;

namespace Blackjack21.Game.Exceptions
{

    /// <summary>
    /// Throws an exception if the hand cannot be split
    /// </summary>
    public class InvalidHandSplitException : Exception
    {

        public InvalidHandSplitException(string message = "Invalid hand split, conditions to split the hand have not been met") : base(message)
        {

        }
    }
}
